gcloud-java-dns/src/main/java/com/google/gcloud/dns/Dns.java

```diff
@@ -0,0 +1,537 @@
+     * continue.
+     */
+    public static ZoneListOption pageToken(String pageToken) {
+      return new ZoneListOption(DnsRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Restricts the list to only zone with this fully qualified domain name.
+     */
+    public static ZoneListOption dnsName(String dnsName) {
+      StringBuilder builder = new StringBuilder();
+      return new ZoneListOption(DnsRpc.Option.DNS_NAME, dnsName);
+    }
+
+    /**
+     * The maximum number of zones to return per RPC.
+     *
+     * <p>The server can return fewer zones than requested. When there are more results than the
+     * page size, the server will return a page token that can be used to fetch other results.
+     */
+    public static ZoneListOption pageSize(int pageSize) {
+      return new ZoneListOption(DnsRpc.Option.PAGE_SIZE, pageSize);
+    }
+  }
+
+  /**
+   * Class for specifying project options.
+   */
+  class ProjectOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 6817937338218847748L;
+
+    ProjectOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the project's fields to be returned by the RPC call.
+     *
+     * <p>If this option is not provided all project fields are returned. {@code
+     * ProjectOption.fields} can be used to specify only the fields of interest. Project ID is
+     * always returned, even if not specified. {@link ProjectField} provides a list of fields that
+     * can be used.
+     */
+    public static ProjectOption fields(ProjectField... fields) {
+      return new ProjectOption(DnsRpc.Option.FIELDS, ProjectField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying change request field options.
+   */
+  class ChangeRequestOption extends AbstractOption implements Serializable {
+
+    private static final long serialVersionUID = 1067273695061077782L;
+
+    ChangeRequestOption(DnsRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify which fields of {@link ChangeRequest} should be returned by the
+     * service.
+     *
+     * <p>If this option is not provided all change request fields are returned. {@code
+     * ChangeRequestOption.fields} can be used to specify only the fields of interest. The ID of the
+     * change request is always returned, even if not specified. {@link ChangeRequestField} provides
+     * a list of fields that can be used.
+     */
+    public static ChangeRequestOption fields(ChangeRequestField... fields) {
+      return new ChangeRequestOption(
+          DnsRpc.Option.FIELDS,
+          ChangeRequestField.selector(fields)
+      );
+    }
+  }
+
+  /**
+   * Class for specifying change request listing options.
+   */
+  class ChangeRequestListOption extends AbstractOption implements Serializable {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://www.codacy.com/app/mziccard/gcloud-java/file/2317493773/issues/source?bid=3123109&fileBranchId=3203698#l375)